### DISCUSSION TITLE
How to assert JAXB generated classes are not removed in Native mode?

### DISCUSSION THREAD

**Discussion by hakuseki** [quarkusio/quarkus#39577] [Q&A] [answered]
I have +10 different XSD files generating POJOs using maven JAXB plugin and store them in the target directory in different packages.

How can I ensure and be certain they are all within the build when going native?

[a comment by an automated account is not shown]

**Comment by zakkak** [comment, marked as answer]
Hi @hakuseki, 

You can have a look at https://github.com/quarkusio/quarkus/blob/ad91929509c6bc1d81ce99739c6e9dfbc4cf5d84/test-framework/junit5/src/main/java/io/quarkus/test/junit/nativeimage/ClassInclusionReport.java to see how we do this in Quarkus integration tests.

Note: it depends on building with `quarkus.native.enable-reports=true`

If you don't need this kind of testing to be automated, then just build with `quarkus.native.enable-reports=true` and grep for the classes you are interested in in the generated `used_classes_*.txt` file under the `target/*-native-image-source-jar/reports` directory.
